Overview.bs

```diff
@@ -2671,3 +2675,227 @@ to be used by default in most shaper implementations. This list was assembled fr
+
+This appendix provides an example of how a typical IFT font would be processed by the [[#extend-font-subset]] algorithm. In this example the IFT font
+contains a mix of both [[#table-keyed]] and [[#glyph-keyed]] patches.
+
+<b>Initial font</b>: contains the following IFT and IFTX patch mappings. Note: when features and design space sets are unspecified in a subset definition
+they are defaulted to an empty set.
+
+<table>
+  <tr><th>Table = "IFT "</th><th colspan="2">Compatibility ID = 0x0000_0000_0000_0001</th></tr>
+  <tr><th>Subset Definition(s)</th><th>URI</th><th>Format Number</th>
+  <tr>
+    <td>
+      ```
+      code points: { 'a', 'b', ..., 'z' }
+      ```
+    </td>
+    <td>//foo.bar/01.tk</td>
+    <td>
+      2, Table Keyed - Partial Invalidation
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { 'A', 'B', ..., 'Z' }
+      ```
+    </td>
+    <td>//foo.bar/02.tk</td>
+    <td>
+      2, Table Keyed - Partial Invalidation
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { '0', '1', ..., '9' }
+      ```
+    </td>
+    <td>//foo.bar/03.tk</td>
+    <td>
+      2, Table Keyed - Partial Invalidation
+    </td>
+  </tr>
+
+  <tr>
+    <td>
+      ```
+      code points: { 'a', 'b', ..., 'z',
+                     'A', 'B', ..., 'Z' }
+      ```
+    </td>
+    <td>//foo.bar/04.tk</td>
+    <td>
+      2, Table Keyed - Partial Invalidation
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { 'a', 'b', ..., 'z',
+                     'A', 'B', ..., 'Z',
+                     '0', '1', ..., '9' }
+      ```
+    </td>
+    <td>//foo.bar/05.tk</td>
+    <td>
+      2, Table Keyed - Partial Invalidation
+    </td>
+  </tr>
+</table>
+<br/>
+<table>
+  <tr><th>Table = "IFTX"</th><th colspan="2">Compatibility ID = 0x0000_0000_0000_0002</th></tr>
+  <tr><th>Subset Definition(s)</th><th>URI</th><th>Format Number</th>
+  <tr>
+    <td>
+      ```
+      code points: { 'a', 'b', ..., 'm' }
+      ```
+    </td>
+    <td>//foo.bar/01.gk</td>
+    <td>
+      3, Glyph Keyed
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { 'n', 'o', ..., 'z' }
+      ```
+    </td>
+    <td>//foo.bar/02.gk</td>
+    <td>
+      3, Glyph Keyed
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { 'A', 'B', ..., 'M' }
+      ```
+    </td>
+    <td>//foo.bar/03.gk</td>
+    <td>
+      3, Glyph Keyed
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { 'N', 'O', ..., 'Z' }
+      ```
+    </td>
+    <td>//foo.bar/04.gk</td>
+    <td>
+      3, Glyph Keyed
+    </td>
+  </tr>
+  <tr>
+    <td>
+      ```
+      code points: { '0', '1', ..., '9' }
+      ```
+    </td>
+    <td>//foo.bar/05.gk</td>
+    <td>
+      3, Glyph Keyed
+    </td>
+  </tr>
+</table>
+
+<b>Optimized Extension Example</b>
+
+Note: this example execution is described as it would be implemented in an optimized client which aims to reduce the number of round trips needing to be
```

Comment:
I think one aspect of this description needs updating in light of #238 (as expected) and the second isn't accurate relative to the example. This is will now be following the selection criteria rather than "making choices". And I don't think it's doing any speculative loading -- it's choosing 3 as it should. Or maybe that's what was meant. 

I think going forward we can talk about the algorithm now that it's nailed down, and occasionally talk about speculative loading. For the latter I want to get away from the term "preload" because we're using that for adding patches to a file on the server side.

Reply:
I've updated the later step that talks about invalidating patch selection to reflect the changes in #238. Also as suggested changed preload to speculative.

For this particular note the speculative loading I'm talking about is for the glyph keyed patches (so not related to invalidating patch selection), I added some text to clarify that.

Reply:
> For this particular note the speculative loading I'm talking about is for the glyph keyed patches (so not related to invalidating patch selection), I added some text to clarify that.

The set requires 'f' and 'P' and we load 01.gk and 04.gk, so I don't see any speculative loading for those.

Reply:
These are speculative in the sense that we have to apply the table keyed patch first and we don't know for sure exactly what it will do. Although unlikely it could modify the entry list for the glyph keyed patches and remove 01.gk and 04.gk entries making the loads for those two unnecessary. Maybe speculative isn't quite the right word here, how about preemptive?

Reply:
OK, hmm. From the text further down:

> The table keyed patch //foo.bar/04.tk is partially invalidating which means the two glyph keyed patches will be remain valid after the application of //foo.bar/04.tk. Thus the client can expect that the two glyph keyed patches will be required in future iterations and preemptively begin the loads for those two patches.

What's ambiguous about the situation such that either "speculative" or "preemptive" really applies? If 04.tk *could* make the other two patches redundant, it would be one thing, but it can't. 

I suppose loading patches in parallel can be seen as an "optimization" but in practice it will kill performance not to, so I don't think we should present it as "optional" except in the loosest sense.

Why is there any linkage between the two patch groups in the algorithm in the first place? If 01.gk arrives before 04.tk, can't you (in terms of correctness) apply it first?

Reply:
Yes, the invalidation modes guarantees that the glyph keyed patches in this case are always safe to apply. However, the invalidation rules don't prevent a table keyed patch from rewriting the glyph keyed map to point to a different set of URIs that still use the same compatibility base (eg. how we handle variable axis extension in the prototype). So there's no harm in applying the glyph keyed patches early, but it's possible (though unlikely) they may become unnecessary if they are superseded by new patches. Definitely agree that saving the round trip is always worth it even in the odd case where you grab patches you don't end up needing. I'll reword this a bit to tone down the "speculative" nature of this behaviour and just present it as what a good implementation should do.

Reply:
Ok, updated the text. What do you think?